libcodechecker/server/client_db_access_server.py

```diff
@@ -124,31 +142,121 @@ def check_auth_in_request(self):
 
         return success
 
+    def __get_product_name(self):
+        """
+        Get product name from the request's URI.
+        """
+
+        # A standard request from a browser looks like:
+        # http://localhost:8001/[product-name]/#{request-parts}
+        # where the parts are, e.g.: run=[run_id]&report=[report_id]
+        #
+        # Rewrite the "product-name" so that the web-server deploys the
+        # viewer client from the www/ folder.
+
+        # The split array looks like ['', 'product-name', ...].
+        first_part = urlparse.urlparse(self.path).path.split('/', 2)[1]
+        return first_part if first_part not in NON_PRODUCT_NAMES else None
```

Comment:
Would make the command line and the thrift client creation lot more complicated. (*) This list doesn't really need to be maintained, as there is almost no chance that a new subfolder under `www-root` is added, as we already have every "resource folder" (fonts, styles, etc.) present and used.

(*): Right now, due to how URLs work, setting up the API endpoint in Thrift in the web code like `new Thrift("Authentication")` automatically puts it under `/[product-name]/Authentication`. Do we want to make it `/product/[product-name]/service/Authentication` or... something else? It would make the webserver code a lot more complicated, as we need to start actually parsing the active URL there.